compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -2183,7 +2184,7 @@ impl<'a> State<'a> {
                     AsmArg::Options(opts) => {
                         s.word("options");
                         s.popen();
-                        let mut options = vec![];
+                        let mut options: SmallVec<[_; 7]> = SmallVec::new();
```

Comment:
nit: it might make more sense to use `ArrayVec` instead of `SmallVec` here, considering that there can never be more than 7 options.

Reply:
TIL: ArrayVec exists

Reply:
Not sure I want to introduce a new dependency, though.

Reply:
:thinking: afaik pretty printing isn't really used often, so I personally would keep using a `vec` here.

`ArrayVec` is fairly nice, though only rarely necessary imo